src/generic/listctrl.cpp

```diff
@@ -2135,7 +2161,9 @@ void wxListMainWindow::OnPaint( wxPaintEvent &WXUNUSED(event) )
         }
 
         // Draw vertical rules if required
-        if ( HasFlag(wxLC_VRULES) && !IsEmpty() )
+        if ( HasFlag(wxLC_VRULES) && (m_listRulesAlternateColourOnBlank ||
+                                      ( ! m_listRulesAlternateColourOnBlank
+                                        && !IsEmpty() ) ) )
```

Comment:
This condition is too complicated, it would be more clear with a temporary variable.